### PR TITLE
feat: load rmrk2 images on item-explore

### PR DESCRIPTION
**Pull request by preschian**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] related #5183
- [x] ~~load cf-image directly if exists~~
- [x] remove old explore-item

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [ ] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [ ] I've tested it at </bsx/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=DY4SQF2iD456tH89aQtz5wv1EV3BbSW8wKKuMcwbmXaj1pM)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

Screenshot 2023-03-07 at 17 55 06

### CODE REVIEW

File: libs/ui/src/components/MediaItem/type/ImageMedia.vue [hunk 1]
```diff
@@ -14,12 +27,37 @@
...
+  if (props.src) {
+    return (
+      props.src.replace(
+        /https:\/\/image(-beta)?.w.kodadot.xyz\/ipfs\//,
+        'https://imagedelivery.net/jk5b6spi_m_-9qC4VTnjpg/'
+      ) + '/public'
+    )
+  }
+
+  return ''
```
Comment (roiLeo) [inline libs/ui/src/components/MediaItem/type/ImageMedia.vue:57]: What's this? can we do without empty return?
is this a good idea to add vue-use in `libs/ui` for me it's supposed to be a bare component

File: components/items/ItemsGrid/ItemsGridImage.vue [hunk 2]
```diff
@@ -37,6 +43,17 @@ onBeforeMount(async () => {
...
   )) as NFTWithMetadata
+
+  if (urlPrefix.value === 'rmrk2' && props.nft.resources.length) {
+    const { src, thumb } = props.nft.resources[0]
+    console.log(src, thumb)
+    item.value = {
+      ...props.nft,
+      image: sanitizeIpfsUrl(thumb || src || ''),
+    }
+    return
```
Comment (roiLeo) [inline components/items/ItemsGrid/ItemsGridImage.vue:54]: :/

Comment (roiLeo) [reply components/items/ItemsGrid/ItemsGridImage.vue:61]: 40 LOC, IMO we should make it a composable or check it before it come through this component.
And empty return statement leaves me a little confused

File: libs/ui/package.json [hunk 3]
```diff
@@ -20,6 +20,7 @@
   },
   "dependencies": {
     "@google/model-viewer": "^1.12.1",
+    "@vueuse/core": "^9.13.0",
```
Comment (roiLeo) [inline libs/ui/package.json:23]: maybe remove if it's not used

File: components/items/ItemsGrid/ItemsGridImage.vue (hunk 2 above)
Comment (vikiival) [reply components/items/ItemsGrid/ItemsGridImage.vue:61]: > And empty return statement leaves me a little confused

Yes please this good thing to introduce some undefined behaviour

File: components/items/ItemsGrid/ItemsGridImage.vue [hunk 4]
```diff
@@ -1,61 +1,30 @@
...
-      v-if="item"
-      class="is-hidden-tablet"
-      :nft="item"
-      :prefix="urlPrefix"
-      :show-price="Number(item?.price) > 0"
-      variant="minimal" />
-  </div>
+  <NeoNftCard
+    v-if="item"
+    class="is-hidden-mobile"
```
Comment (daiagi) [inline components/items/ItemsGrid/ItemsGridImage.vue:4]: render nothing on mobile devices?

image

Comment (daiagi) [reply components/items/ItemsGrid/ItemsGridImage.vue]: I agree, i think we are over using this approach and a util/composable for easy access to screen dimensions
and change props accordingly would be better for some cases

File: components/items/ItemsGrid/ItemsGridImage.vue [hunk 5]
```diff
@@ -1,61 +1,29 @@
...
-  const animation_url = sanitizeIpfsUrl(metadata.animation_url || '')
-
-  item.value = {
-    ...props.nft,
-    name: props.nft.name || metadata.name || props.nft.id,
-    image,
-    animation_url,
-    type: metadata.type || '',
-  }
+  item.value = await getNftMetadata(props.nft, urlPrefix.value)
```
Comment (daiagi) [inline components/items/ItemsGrid/ItemsGridImage.vue:27]: thinking out loud here...
would it be better to have getNftMetadata return a ref that is being filled out when the internal promise is resolved?
this way you would not need to call it in onBeforeMount
like we do in useGalleryItem and other places

File: components/items/ItemsGrid/ItemsGridImage.vue (hunk 4 above)
Comment (daiagi) [reply components/items/ItemsGrid/ItemsGridImage.vue]: and btw... change the name of ```isMobileDevice``` in ```@/utils/extension``` to something like ```isTouchScreen```

**Review by daiagi — APPROVED** [review]
nice performance boost with the lazy loading :+1:

File: pages/_prefix/explore/items.vue [hunk 6]
```diff
@@ -1,8 +1,5 @@
 <template>
-  <div>
-    <Items v-if="redesign" />
-    <Gallery v-else class="container" />
-  </div>
+  <Items />
```
Comment (roiLeo) [inline pages/_prefix/explore/items.vue:2]: is it done already?
can you wrap `Items` with `<div>`? (might be needed to add container here #5089)

File: package.json [hunk 7]
```diff
@@ -76,10 +76,10 @@
...
     "@paraspell/sdk": "0.0.25",
     "@pinia/nuxt": "^0.4.7",
-    "@polkadot/extension-dapp": "^0.44.8",
-    "@polkadot/extension-inject": "^0.44.8",
-    "@polkadot/ui-keyring": "^2.11.1",
-    "@polkadot/vue-identicon": "^2.11.1",
+    "@polkadot/extension-dapp": "^0.44.9",
+    "@polkadot/extension-inject": "^0.44.9",
+    "@polkadot/ui-keyring": "^2.12.1",
+    "@polkadot/vue-identicon": "^2.12.1",
```
Comment (roiLeo) [inline package.json:82]: alright if nothing break that's cool

**Review by roiLeo — COMMENTED** [review]
> I also lazy load the collection popover. just found out that we fetch so many requests on collection popover

Wow 😮 It looks very promising!

File: pages/_prefix/explore/items.vue (hunk 6 above)
Comment (roiLeo) [reply pages/_prefix/explore/items.vue]: Yes, I'm ok with that (https://github.com/kodadot/nft-gallery/issues/5069#issuecomment-1458381445)

**Comment by vikiival** [comment]
pay 70 usd

**Comment by vikiival** [comment]
Please resolbe conflicts @preschian

**Pull request merged**